tests/aws/services/sqs/test_sqs.py

```diff
@@ -2802,6 +2804,80 @@ def test_dead_letter_queue_max_receive_count(
             == result_send["MessageId"]
         )
 
+    @markers.aws.validated
+    def test_dead_letter_queue_message_attributes(
+        self,
+        aws_client,
+        sqs_create_queue,
+        sqs_get_queue_arn,
+        snapshot,
+    ):
+        sqs = aws_client.sqs
+
+        queue_name = f"queue-{short_uid()}"
+        dead_letter_queue_name = f"dl-queue-{short_uid()}"
+        dl_queue_url = sqs_create_queue(QueueName=dead_letter_queue_name)
+        dl_queue_arn = sqs_get_queue_arn(dl_queue_url)
+        queue_url = sqs_create_queue(
+            QueueName=queue_name,
+            Attributes={
+                "RedrivePolicy": json.dumps(
+                    {"deadLetterTargetArn": dl_queue_arn, "maxReceiveCount": 1}
+                )
+            },
+        )
+
+        snapshot.match("dlq-arn", dl_queue_arn)
+        snapshot.match("sourcen-arn", sqs_get_queue_arn(queue_url))
+
+        # check that attributes are retained
+        msg_attrs = {"MyAttribute": {"StringValue": "foobar", "DataType": "String"}}
+        msg_system_attrs = {
+            "AWSTraceHeader": {
+                "StringValue": "Root=1-5759e988-bd862e3fe1be46a994272793;Parent=53995c3f42cd8ad8;Sampled=1",
+                "DataType": "String",
+            }
+        }
+        # send a messages
```

Comment:
nit: typo

Reply:
i want to do a spring cleaning of the SQS code base (including comment formatting, doc strings, maybe some light refactoring), and will clean up the comments as well :-)